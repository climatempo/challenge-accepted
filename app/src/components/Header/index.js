import React from 'react';

import { Container, Logo } from './styles';

import logo from '../../assets/img/logo-white.png';

export default function Header() {
	return (
		<Container>
			<Logo source={logo} />
		</Container>
	);
}
