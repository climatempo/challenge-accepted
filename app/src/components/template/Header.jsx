import React from 'react'

import Logo from './Logo'

export default props =>
    <React.Fragment>
        <header id="header">
            <div className="inner">
                <Logo />
            </div>
        </header>
    </React.Fragment>