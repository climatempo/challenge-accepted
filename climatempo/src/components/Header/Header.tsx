import React from "react";
import styled from "styled-components";
import logo from "../../assets/logo-white.png";

type TypeHeader = {
  exit: () => void;
};

function Header({ exit }: TypeHeader) {
  return (
    <Container>
      <Logo src={logo} alt="logo" />
      <ButtonExit onClick={exit}>Sair</ButtonExit>
    </Container>
  );
}

const Container = styled.div`
  width: 100vw 100%;
  height: 70px;
  background: #04a1cc;
  border-radius: 0px 0px 50px 50px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 10px 40px;
`;

const Logo = styled.img`
  width: 188px;
  height: 28px;

  @media (max-width: 888px) {
    width: 144px;
    height: 18px;
  }
`;

const ButtonExit = styled.button`
  width: 80px;
  height: 30px;
  background: #ffffff;
  border-radius: 50px;
  border: none;
  cursor: pointer;
  font-family: "Roboto", sans-serif;
  font-style: normal;
  font-weight: 700;
  font-size: 18px;
  line-height: 29px;
  /* identical to box height */
  color: #047ab2;

  :hover {
    background: transparent;
    border: 1px solid #ffffff;
    color: #ffffff;
  }

  @media (max-width: 888px) {
    width: 70px;
    height: 25px;
    font-size: 16px;
    display: flex;
    justify-content: center;
    align-items: center;
  }
`;

export default Header;
