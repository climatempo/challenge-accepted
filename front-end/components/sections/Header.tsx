import { Box, Button, Container, Input, Stack } from '@chakra-ui/react'
import Image from 'next/image'
import HeaderImage from '../../public/assets/logo-climatempo-stormgeo.svg'
import React from 'react'

export const Header = () => {
    return (
        <Box bg="blue.400" py="5">
            <Container maxW="container.xl">
                <Stack direction="row" justify="space-between" align="center" >
                    <Image src={HeaderImage} />
                    <Input type="text" size="md" placeholder="Procure por sua cidade..." flexBasis="50%"/>
                    <Button>Menu</Button>
                </Stack>
            </Container>
        </Box>
    )
}