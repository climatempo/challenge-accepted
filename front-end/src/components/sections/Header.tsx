import { Box, Button, Container, Image, Stack, useDisclosure, useMediaQuery } from '@chakra-ui/react'
import HeaderImage from '../../assets/logo-climatempo-stormgeo.svg'
import React, { useEffect } from 'react'
import { AutoCompleteSearch } from '../ui/AutoCompleteSearch'
import { ILocale } from '../../types/interfaces'
import { GET_LOCALES } from '../../graphql/queries'
import { useQuery } from '@apollo/client'
import { Link } from 'react-router-dom'
import { ModalPreferences } from '../ui/PreferencesModal'

/**
 * Componente do header do site, com campo de busca e botões de opções
 * @returns Componente do cabeçalho da página
 */
export const Header = () => {
    const { data } = useQuery<{ weathers: Array<{ locale: ILocale }> }>(GET_LOCALES);
    const [locales, setLocales] = React.useState<ILocale[]>()
    const [isMobile] = useMediaQuery("(max-width: 465px)")
    const { isOpen, onOpen, onClose } = useDisclosure()

    useEffect(() => { setLocales(data?.weathers.map(locale => locale.locale)) }, [data])
    return (
        <>
            <Box bg="blue.400" py="5">
                <Container maxW="container.xl">
                    <Stack direction="row" justify="space-between" align="center" >
                        <Link to="/"><Image src={HeaderImage} /></Link>
                        {isMobile ? '' : <AutoCompleteSearch suggestions={locales!} />}
                        <Button onClick={onOpen}>Menu</Button>
                    </Stack>
                </Container>
                <ModalPreferences isOpen={isOpen} onClose={onClose} onOpen={onOpen}></ModalPreferences>
            </Box>
            {isMobile ?
                <Box margin="auto" mt="6">
                    <AutoCompleteSearch suggestions={locales!} />
                </Box> : ''}
        </>
    )
}