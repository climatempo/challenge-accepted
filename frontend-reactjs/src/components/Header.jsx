import React from 'react';
import styled from 'styled-components';

const Header = styled.header`
  justify-content: center;
  align-items: center;
  height: 55px;
  background-color: #1665c0;
`;

export default () => (
  <Header>
    <img
      src={`${window.location.origin}/images/logo-white.png`}
      alt=""
      style={{ height: '18px' }}
    />
  </Header>
);
