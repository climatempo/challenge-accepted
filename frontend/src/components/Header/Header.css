.Header {
    background-color: #1178f2;
    height: 4rem;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
}

.Image {
    width: 9rem;
}