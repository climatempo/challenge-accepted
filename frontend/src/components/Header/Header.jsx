import React from "react";
import LogoClima from "../images/logo-clima.png";
import "./Header.css";

export default function Header() {
    return(
        <header className="Header">
            <img className="Image" src={LogoClima} alt="logo climatempo" />
        </header>
    );
};