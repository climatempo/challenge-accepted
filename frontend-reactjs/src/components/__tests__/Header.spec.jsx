import React from 'react';
import toJson from 'enzyme-to-json';

import { mount } from '../../tests-setup';

import Header from '../Header';

describe('Header Component', () => {
  let wrapper = null;

  beforeEach(() => {
    wrapper = mount(<Header />);
  });

  it('should correctly render', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
