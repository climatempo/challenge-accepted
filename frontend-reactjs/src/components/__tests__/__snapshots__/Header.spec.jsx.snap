// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should correctly render 1`] = `
<_default>
  <styled.header>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "ixnIfc",
            "rules": Array [
              "
  justify-content: center;
  align-items: center;
  height: 55px;
  background-color: #1665c0;
",
            ],
          },
          "displayName": "styled.header",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "header",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <header
        className="sc-bdVaJa ixnIfc"
      >
        <img
          alt=""
          src="http://localhost/images/logo-white.png"
          style={
            Object {
              "height": "18px",
            }
          }
        />
      </header>
    </StyledComponent>
  </styled.header>
</_default>
`;
